### PR TITLE
Add release pipeline

### PR DESCRIPTION
**Pull request by hovaesco**
ref: https://github.com/dbt-labs/dbt-presto/blob/master/.github/workflows/release.yml

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.9.5"
+
+      - name: Test release
```
Comment (findinpath) [inline .github/workflows/release.yml:22]: what exactly is being tested here?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        env:
+          TWINE_USERNAME: __token__
+          TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+        run: |
+          python3 -m venv env
+          source env/bin/activate
+          pip install -r dev_requirements.txt
+          pip install twine wheel setuptools
+          python setup.py sdist bdist_wheel
+          twine upload --non-interactive dist/dbt_trino-${{env.version_number}}-py3-none-any.whl dist/dbt-trino-${{env.version_number}}.tar.gz
```
Comment (findinpath) [inline .github/workflows/release.yml:111]: do you think it would make sense to release to https://test.pypi.org/ as well?

**Review by findinpath — APPROVED** [review]

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
+name: dbt-trino release
+
+on:
+  workflow_dispatch:
+
+# Release version number that must be updated for each release
+env:
+  version_number: "0.21.0"
```
Comment (findinpath) [inline .github/workflows/release.yml:8]: There are now several places to update before doing a release.

Could you please document throughly the release process for eventual new maintainers?

File: .github/workflows/release.yml (hunk 1 above)
Comment (hovaesco) [reply .github/workflows/release.yml:22]: test PyPi package before uploading

File: .github/workflows/release.yml (hunk 3 above)
Comment (hovaesco) [reply .github/workflows/release.yml:8]: I was looking at the pipeline again and I think that the whole approach in bumping version in here is wrong. 

Version bump is done just before doing a release to Github, and it's never tested.. tests needs to be done using new dbt version.

I need to redesign the release pipeline.

Comment (hovaesco) [reply .github/workflows/release.yml:8]: I will describe the release process in README

**Comment by hovaesco** [comment]
Bumping version and testing has been split from the release pipeline.

File: README.md [hunk 4]
```diff
@@ -254,6 +254,22 @@ or
...
+
+Before doing a release dbt's version requires updating.
+In order to bump dbt-trino and dbt-core version run:
+
+```sh
+bumpversion --config-file .bumpversion-dbt.cfg patch --new-version <new-version>
+bumpversion --config-file .bumpversion.cfg patch --new-version <new-version> --allow-dirty
+```
+
+Additionally bump dbt-core version in dbt testing image `docker/dbt/Dockerfile`.
```
Comment (findinpath) [inline README.md:267]: Can we do all these steps  bumping steps with only *1* command?

File: README.md [hunk 5]
```diff
@@ -254,6 +254,22 @@ or
...
+
+Next step is to merge bump commit and making sure that test suite pass.
+
+Finally to release `dbt-trino` to PyPi and GitHub trigger release workflow `release.yml`.
```
Comment (findinpath) [inline README.md:271]: How does a maintainer trigger the "release.yml" workflow?

**Comment by findinpath** [comment]
I find these changes very welcome. I am looking forward to see them in action with the next dbt release.

File: README.md (hunk 5 above)
Comment (hovaesco) [reply README.md:271]: Via GHA: Actions --> release --> Run workflow. It's triggered using `workflow_dispatch` event type. https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows#workflow_dispatch

File: README.md (hunk 4 above)
Comment (hovaesco) [reply README.md:267]: With bumpversion I think it cannot be done. We can have a script which does all the above. But I don't think that is a big deal to update it in one more place.

Comment (findinpath) [reply README.md:267]: 👍

**Comment by hovaesco** [comment]
It's going to be tested while releasing dbt-trino v1.0.0.

**Pull request merged**